src/style-property-map-readonly.js

```diff
@@ -36,13 +36,12 @@
   };
 
   StylePropertyMapReadOnly.prototype.getProperties = function() {
-    var output = [];
-    for (var i = 0, l = this._styleObject.length; i < l; ++i) {
-      var property = this._styleObject[i];
-      // TODO: Construct the objects for each type of thing and add them to the
-      // output array.
+    var cssPropertiesFullText = this._styleObject.cssText.split('; ');
+    var cssProperties = [];
+    for (var i = 0; i < cssPropertiesFullText.length; i++) {
+      cssProperties[i] = cssPropertiesFullText[i].split(':')[0];
```

Comment:
Sorry I didn't notice this before. It'd be more efficient to avoid parsing this string and doing what you did in `StylePropertyMap.prototype.getProperties`.

Also, if you move that getProperties to here, you don't need the one in StylePropertyMap since it inherits from this one.